### PR TITLE
Path movement first iteration done, player added, enemy destroyed on …

### PR DESCRIPTION
**Pull request by krooq**
…collision with anything but other enemies

### CODE REVIEW

**Comment by JPGalovic** [comment]
[Feature - Player Movement: Draw path to move.](https://trello.com/c/Yyv4TUfK/3-feature-player-movement-draw-path-to-move)

File: Assets/Scripts/PathMovement.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class PathMovement : MonoBehaviour {
+
+    public GameObject thingToMove;
+    private List<Vector3> path = new List<Vector3>();
```
Comment (xxfast) [inline Assets/Scripts/PathMovement.cs:9]: This probably works best as a queue

File: Assets/Scripts/PathMovement.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		Physics.Raycast(ray, out rayHit);
+		Debug.Log(rayHit.collider.gameObject);
+		return rayHit.collider.gameObject;
+	}
+
+    // Need to factor in the total length of ppath so that
+    // 1. there is a cap to path length
+    // 2. the movement is uniform speed no matter the path length
+    public void moveToNextPointOnPath() {
+        if(nextPointIndex < 0) {
```
Comment (xxfast) [inline Assets/Scripts/PathMovement.cs]: `nextPointIndex` would be redundant if a queue was used

**Review by xxfast — APPROVED** [review]
Working as intended, could be further optimized further using a Queue

**Pull request merged**